feat(oxc_transformer): support jsx pragma that are long member expressions

IWANABETHATGUY opened the pull request:
related tests: 

https://github.com/evanw/esbuild/blob/d34e79e2a998c21bb71d57b92b0017ca11756912/internal/bundler_tests/bundler_default_test.go#L6188-L6236

IWANABETHATGUY commented:
* **#7538** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7538?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7538?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (commented):
Can you provide a minimal example? I'm not sure what we are changing here and what we are testing against.

Test can be added to tasks/transform_conformance/tests/babel-plugin-transform-react-jsx/test/fixtures

IWANABETHATGUY commented:
> Can you provide a minimal example? I'm not sure what we are changing here and what we are testing against.
> 
> Test can be added to tasks/transform_conformance/tests/babel-plugin-transform-react-jsx/test/fixtures

TLDR, 
before we could only support `pragma` like `a.b` (length could only be two if it is memberExpr),  or `a` a identifier.

IWANABETHATGUY commented on tasks/transform_conformance/tests/babel-plugin-transform-react-jsx/test/fixtures/misc/arbitrary_length_member_expr/options.json:
```diff
@@ -0,0 +1,11 @@
+      "transform-react-jsx",
+      {
+        "runtime": "classic",
+        "pragma": "a.b.c"
```
Before an error will be raised, since before we only support member expr length of two, https://github.com/oxc-project/oxc/blob/6ef9cb06a3b56ee35bdffd1d9487ddb6c8746c73/crates/oxc_transformer/src/jsx/jsx_impl.rs?plain=1#L340-L345

Boshen commented:
I'll let the code owners to review.

IWANABETHATGUY commented:
any feedback would be good, this pr intent to fix `jsx.factory` issue in rolldown

overlookmotel commented:
It's unfortunate that we have to allocate a `Vec` for the default, but I guess no easy way around that.

overlookmotel commented:
### Merge activity

* **Dec 2, 9:01 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

IWANABETHATGUY committed:
feat(oxc_transformer): support jsx pragma that are long member expressions (#7538)

related tests:

https://github.com/evanw/esbuild/blob/d34e79e2a998c21bb71d57b92b0017ca11756912/internal/bundler_tests/bundler_default_test.go#L6188-L6236
